src/Razor/test/Microsoft.VisualStudio.Razor.IntegrationTests/HtmlCodeActionsTests.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.VisualStudio.Razor.IntegrationTests
+{
+    public class HtmlCodeActionsTests : AbstractRazorEditorTest
+    {
+        [IdeFact]
+        public async Task HtmlCodeActionsTests_RemoveTag()
+        {
+            // Open the file
+            await TestServices.SolutionExplorer.OpenFileAsync(RazorProjectConstants.BlazorProjectName, RazorProjectConstants.CounterRazorFile, ControlledHangMitigatingCancellationToken);
```

Comment:
LOVE that we're adding more integration tests.

When possible/sensible it's good for us to include multiple scenarios in a single `IdeFact` just because of the overhead involved in separate tests. Up to you if that applies in this case, our standards will probably have to get tighter as we add more tests though.

Reply:
I definitely did it because I am concerned about testing methodology, and not just because it's much easier to write them 😁

Not sure I agree about the multiple scenarios necessarily though... In this case, to combine these two tests means closing one file, opening another, then I'd have to make sure I wait for that new file to be ready etc. Might save a few seconds, but feels like it would introduce flakiness.

I guess I could have set some custom file text for the whole thing, and cover both cases in one file though...

Reply:
> I guess I could have set some custom file text for the whole thing, and cover both cases in one file though...

Yeah, in this instance that would be my suggestion. But again, not a blocker. The other unit tests don't follow this suggestion terribly well, it's more of a future-proofing thing.